src/main/java/knightminer/inspirations/common/ClientProxy.java

```diff
@@ -54,11 +57,22 @@ protected static void registerItemColors(ItemColors itemColors, IItemColor handl
 		}
 	}
 
-	protected static void replaceTexturedModel(ModelBakeEvent event, ModelResourceLocation location, String key, boolean item) {
+	protected static void replaceModel(ModelBakeEvent event, ModelResourceLocation location, BiFunction<IBakedModel, IModel, IBakedModel> modelMaker) {
 		IModel model = ModelLoaderRegistry.getModelOrLogError(location, "Error loading model for " + location);
 		IBakedModel standard = event.getModelRegistry().get(location);
-		IBakedModel finalModel = new TextureModel(standard, model, DefaultVertexFormats.BLOCK, key, item);
-
+		IBakedModel finalModel = modelMaker.apply(standard, model);
 		event.getModelRegistry().put(location, finalModel);
 	}
+
+	protected static void replaceTexturedModel(ModelBakeEvent event, ModelResourceLocation location, String key, boolean item) {
+		replaceModel(
+				event, location,
+				(orig, model) -> new TextureModel(orig, model, item ? DefaultVertexFormats.ITEM : DefaultVertexFormats.BLOCK, key, item)
+		);
+	}
+
+	protected static void replaceBothTexturedModels(ModelBakeEvent event, ResourceLocation loc, String key) {
```

Comment:
I would suggest an overload for this which takes a block instead of a resource location as the second parameter.